servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```
@@ -0,0 +1,534 @@
+                                       ", servletClass: " + servletClass, e);
+        }
+    }
+
+    @Override
+    public <T extends Servlet> T createServlet(Class<T> clazz) throws ServletException {
+        try {
+            requireNonNull(clazz, "clazz");
+            return clazz.getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            throw new ServletException("Failed to create a servlet: " + clazz.getSimpleName(), e);
+        }
+    }
+
+    @Override
+    @Nullable
+    public DefaultServletRegistration getServletRegistration(String servletName) {
+        requireNonNull(servletName, "servletName");
+        return servletRegistrations.get(servletName);
+    }
+
+    @Override
+    public Map<String, DefaultServletRegistration> getServletRegistrations() {
+        return servletRegistrations;
+    }
+
+    @Override
+    public FilterRegistration.Dynamic addFilter(String filterName, String className) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public FilterRegistration.Dynamic addFilter(String filterName, Filter filter) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public FilterRegistration.Dynamic addFilter(String filterName, Class<? extends Filter> filterClass) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public <T extends Filter> T createFilter(Class<T> clazz) throws ServletException {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Nullable
+    @Override
+    public FilterRegistration getFilterRegistration(String filterName) {
+        requireNonNull(filterName, "filterName");
+        return null;
+    }
+
+    @Override
+    public Map<String, ? extends FilterRegistration> getFilterRegistrations() {
+        return ImmutableMap.of();
+    }
```

I think we should throw UnsupportedOperationException here. Let me fix this.